adobe/spectrum-web-components
# feat: add Swatch and Swatch Group patterns

## Description
Add [Swatch](https://opensource.adobe.com/spectrum-css/swatch.html) and [Swatch Group](https://opensource.adobe.com/spectrum-css/swatchgroup.html) patterns via the `@spectrum-web-components/swatch` package.
- expand `base` to support the `when` directive

## Related issue(s)
- fixes #2380

## Motivation and context
Moar Spectrum!

## How has this been tested?
    1. Go [here](https://swatch--spectrum-web-components.netlify.app/components/swatch/)
    2. Learn about the capabilities of a Swatch
    1. Go [here](https://swatch--spectrum-web-components.netlify.app/components/swatch-group/)
    2. Learn about the capabilities of a Swatch Group
    1. Go [here](https://swatch--spectrum-web-components.netlify.app/storybook/index.html?path=/story/swatch-group--default)
    2. See some interactive examples of the Swatch Group
    1. Go [here](https://swatch--spectrum-web-components.netlify.app/storybook/index.html?path=/story/swatch--default)
    2. See some interactive examples of the Swatch

## Screenshots (if appropriate)
<img width="827" alt="image" src="https://user-images.githubusercontent.com/1156657/176248487-f7bcd84a-ce6b-4608-ba13-e6450f7bd75f.png">

## Types of changes

## Checklist
